lib/repl.js

```diff
@@ -1507,72 +1508,60 @@ function regexpEscape(s) {
 
 // If the error is that we've unexpectedly ended the input,
 // then let the user try to recover by adding more input.
+// Note: `e` (the original exception is not used by the current implemention,
+// but may be needed in the future.)
 function isRecoverableError(e, code) {
-  if (e && e.name === 'SyntaxError') {
-    var message = e.message;
-    if (message === 'Unterminated template literal' ||
-        message === 'Unexpected end of input') {
-      return true;
-    }
+  let recoverable = false;
+
+  // Determine if the point of the any error raised is at the end of the input.
+  // There are two cases to consider:
+  //   1.  Any error raised after we have encountered the 'eof' token.
+  //       This prevents us from declaring partial tokens (like '2e') as
+  //       recoverable.
+  //   2.  Three cases where tokens can legally span lines.  This is
+  //       template, comment, and strings with a backslash at the end of
+  //       the line, indicating a continuation.
+  acorn.plugins.repl = (parser) => {
+    parser.extend('nextToken', (nextToken) => {
+      return function(pos, message) {
+        nextToken.call(this, pos, message);
+
```

Comment:
For my education, can I ask why?  First, I don't see `const ReflectApply = Reflect.apply` as a common pattern inside of the node codebase (I only found two occurrences, and even those two differed).  Second, I believe that `.call` is more concise and clearer (at a minimum, I wouldn't need to put `[]` around the arguments).  What am I missing?

Reply:
Since core is not run in its own isolated context but in the one shared with the user, its builtins are the same ones that user-land can touch and interact with. Grabbing references to [builtin prototype methods](https://github.com/nodejs/node/blob/85b0f1649d07294afea6cfbb3082c18a3dc2079a/lib/internal/modules/esm/create_dynamic_module.js#L6-L7) and [static builtin methods](https://github.com/nodejs/node/blob/85b0f1649d07294afea6cfbb3082c18a3dc2079a/lib/internal/bootstrap/loaders.js#L59) before user-land code executes is an attempt to strengthen the core against accidental user-land augmentation. While handled inconsistently in Node core, new code added seems to have these guards. It's reasonable, where possible, to keep that up.

The use of `ReflectApply` covers the `Function#call` case.

Reply:
@jdalton there was a long discussion around this but no conclusion. As long as this is not something everyone agrees to I think it is best to keep these things to the PR openers.